src/pages/History/index.tsx

```diff
@@ -36,27 +35,27 @@ export const HistoryUi: React.FC<IHistoryUiProps> = ({
       setShowOnlyMyTxns={setShowOnlyMyTxns}
       walletConnected={walletConnected}
     />
-    <HistoryTable transactions={transactions} isFetching={isFetching} />
-    {total && (
+    <HistoryTable transactions={transactions} isLoading={isLoading} />
+    {total ? (
       <Pagination
         itemsCount={total}
         onChange={(nextIndex: number) => {
-          setCurrentPage(nextIndex + 1);
+          setCurrentPage(nextIndex);
           window.scrollTo(0, 0);
         }}
         itemsPerPageCount={limit || 20}
       />
-    )}
+    ) : null}
   </div>
 );
 
 const History: React.FC = () => {
   const { account } = useContext(AuthContext);
   const accountAddress = account?.address;
-  const [currentPage, setCurrentPage] = useState(1);
+  const [currentPage, setCurrentPage] = useState(0);
   const [eventType, setEventType] = useState<TransactionEvent | typeof ALL_VALUE>(ALL_VALUE);
   const [showOnlyMyTxns, setShowOnlyMyTxns] = useState(false);
-  const { data: { transactions, total, limit } = { transactions: [] }, isFetching } =
+  const { data: { transactions, total, limit } = { transactions: [] }, isLoading } =
```

Comment:
So I get why you renamed `isFetching` to `isLoading` on the `Table` component, however I think in this case passing it `isFetching` is the right approach.
When the page changes, displaying the loading indicator is great because it shows users something is happening. Otherwise, when I press on a button in the navigation nothing really happens and the current data doesn't change until the new data has been loaded: as a user I could think this is a bug.